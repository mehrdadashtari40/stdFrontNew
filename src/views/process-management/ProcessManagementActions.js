import getAuthonticatedJSON from '../../common/utils/functions/getAuthenticatedJSON';
import postAuthenticatedJSON from '../../common/utils/functions/postAuthenticatedJSON';
import { config } from '../../config/config';

export const INSERT_NEW = 'INSERT_NEW';
export const GET_PROCESS_LIST = 'GET_PROCESS_LIST';
export const GET_CATEGORIES = 'GET_CATEGORIES';
export const DEACTIVE_PROCESS = 'GET_CATEGORIES';
export const DELETE_PROCESS = 'DELETE_PROCESS';
export const DELETE_CASES = 'DELETE_CASES';

export function insertNew(url, request) {
	return dispatch => {
		return postAuthenticatedJSON(url, request).then(response => {
			dispatch(setInsertNewData(response));
		});
	};
}

export function setInsertNewData(data) {
	return {
		type: INSERT_NEW,
		data: data,
	};
}

export function getProcessList(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setProcessList(response.processes));
		});
	};
}

export function setProcessList(data) {
	return {
		type: GET_PROCESS_LIST,
		data: data,
	};
}

export function getCategories(url) {
	return dispatch => {
		return getAuthonticatedJSON(url).then(response => {
			dispatch(setCategories(response));
		});
	};
}

export function setCategories(data) {
	return {
		type: GET_CATEGORIES,
		categories: data,
	};
}

export function deactivateProcessCases(url, data,process) {
	return dispatch => {
		return postAuthenticatedJSON(url, data).then(response => {
			return {
				type: DELETE_CASES,
				data: data,
				payload: process,
			};
		});
	};
}

export function deactivateProcess(url, data) {
	return dispatch => {
		return postAuthenticatedJSON(url, data).then(response => {
			if (response === 1) dispatch(setDeactivateProcess(response));
		});
	};
}

export function setDeactivateProcess(data) {
	return {
		type: DEACTIVE_PROCESS,
		data: data,
	};
}

export function deleteProcess(url, data) {
	return dispatch => {
		return postAuthenticatedJSON(url, data).then(response => {
			dispatch(setDeleteProcess(response));
		});
	};
}

export function setDeleteProcess(data) {
	return {
		type: DELETE_PROCESS,
		data: data,
	};
}
