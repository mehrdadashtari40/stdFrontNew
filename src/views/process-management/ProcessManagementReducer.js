import {
	INSERT_NEW,
	GET_PROCESS_LIST,
	GET_CATEGORIES,
	DEACTIVE_PROCESS,
	DELETE_PROCESS, DELETE_CASES,
} from './ProcessManagementActions';

const initialState = {
	data: [],
	categories: [],
};

export function ProcessManagementReducer(state = initialState, action) {
	switch (action.type) {
		case INSERT_NEW:
			return {
				...state,
				data: action.data,
			};

		case GET_PROCESS_LIST:
			return {
				...state,
				data: action.data,
			};
		case GET_CATEGORIES:
			return {
				...state,
				categories: action.categories,
			};
		case DEACTIVE_PROCESS:
			return {
				...state,
				data: action.data,
			};
		case DELETE_PROCESS:
			return {
				...state,
				data: action.data,
			};
		case DELETE_CASES:
			return {
				...state,
				data: action.payload,
			};
		default:
			return state;
	}
}
