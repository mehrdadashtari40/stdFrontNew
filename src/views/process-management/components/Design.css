
:global(#main) {
    padding-bottom: 0px!important;
}