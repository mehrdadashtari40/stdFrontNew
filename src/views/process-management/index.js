import Design from './components/Design';
export const routes = [
	{
		path: "/process-management",
		exact: true,
		component: Design,
		name: 'Design',
	},
];
export * from './ProcessManagementActions';
export * from './ProcessManagementReducer';
