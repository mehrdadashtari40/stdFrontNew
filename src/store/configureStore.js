import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';

import { userReducer, requestUserInfo } from '../common/user';
import { i18nReducer, i18nInit } from '../common/i18n';
import { trelloReducer } from '../views/trello';
import { systemManagementReducer } from '../views/system-management';
import { ProcessManagementReducer } from '../views/process-management';

import {reducer as layout2Reducer} from '../common/layout/_redux/layoutRedux'
import {reducer as tableReducer} from '../../src/common/tables/_redux/tableRedux'


import {reducer as basicInformationReducer} from '../views/basic-information/_redux/baseInfoRedux';
import LogsReducer from '../views/logs/LogReducesrs';

import BIReducer from '../views/BI/BIReducer';
import {reducer as inboxReducer} from "../views/inbox/_redux/inboxRedux";
import {reducer as authReducer} from '../../src/views/auth/_redux/authRedux'

import {reducer as ServiceDeskReducer} from "../views/service-desk/_redux/serviceDeskRedux";
import {reducer as SupervisorReducer} from "../views/supervisor/_redux/supervisorRedux";
import {reducer as CustomSettingsReducer} from "../views/customSettings/_redux/customSettingsRedux";
import {reducer as MenuReducer} from "../views/menu/_redux/menuRedux";
import {reducer as WorkDeskReducer} from "../views/work-desk/_redux/workDeskRedux";

export const rootReducer = combineReducers({
	auth: authReducer,
	user: userReducer,
	i18n: i18nReducer,
	basicInformation: basicInformationReducer,
	processManagement: ProcessManagementReducer,
	systemManagement: systemManagementReducer,
	ServiceDesk:ServiceDeskReducer,
	BI:BIReducer,
	trello:trelloReducer,
	inbox:inboxReducer,
	layout2:layout2Reducer,
	table:tableReducer,
	logs:LogsReducer,
	supervisor:SupervisorReducer,
	customSettingsReducer:CustomSettingsReducer,
	menuReducer:MenuReducer,
	workDeskReducer:WorkDeskReducer
});

const store = createStore(
	rootReducer,
	applyMiddleware(thunk)
);

store.dispatch(requestUserInfo());

store.dispatch(i18nInit());

export default store;
