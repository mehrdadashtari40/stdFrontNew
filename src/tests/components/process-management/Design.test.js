import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../../store/configureStore';
import configureStore from 'redux-mock-store';
import * as selectActions from '../../../views/process-management/ProcessManagementActions';
import Design from '../../../views/process-management/components/Design';

const mokeData = [
	{
		PRO_UID: '2683031765e9d910cb642c9035051112',
		PRO_TITLE: '1213',
		PRO_DESCRIPTION: '123',
		PRO_PARENT: '2683031765e9d910cb642c9035051112',
		PRO_STATUS: 'ACTIVE',
		PRO_TYPE: 'NORMAL',
		PRO_CATEGORY: '',
		PRO_UPDATE_DATE: null,
		PRO_CREATE_DATE: '1399/02/01 16:39:48',
		PRO_DEBUG: '0',
		PRO_TYPE_PROCESS: 'Public',
		USR_UID: '00000000000000000000000000000001',
		USR_USERNAME: 'admin',
		USR_FIRSTNAME: 'Administrator',
		USR_LASTNAME: 'admin',
		CATEGORY_UID: null,
		CATEGORY_NAME: null,
		PROJECT_TYPE: 'bpmn',
		PRO_CATEGORY_LABEL: '- No Category -',
		PRO_DEBUG_LABEL: 'Off',
		PRO_STATUS_LABEL: 'Active',
		PRO_CREATE_USER_LABEL: 'Administrator admin',
		CASES_COUNT_TO_DO: 0,
		CASES_COUNT_COMPLETED: 0,
		CASES_COUNT_DRAFT: 0,
		CASES_COUNT_CANCELLED: 0,
		CASES_COUNT: 0,
	},
];

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , Design Component', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(wrapper).toMatchSnapshot();
});

it('invokes `componentDidMount` when mounted', () => {
	jest.spyOn(Design.prototype, 'componentDidMount');
	shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	expect(Design.prototype.componentDidMount).toHaveBeenCalled();
	Design.prototype.componentDidMount.mockRestore();
});

it('exportProcess HaveBeenCalled ?', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'exportProcess');

	instance.forceUpdate();

	const Button = wrapper.find('#exportProcess');
	Button.simulate('click');
	expect(spy).toHaveBeenCalled();
});

// it('handleAssignVariable HaveBeenCalled ?', () => {
// 	const wrapper = shallow(
// 		<Provider store={store}>
// 			<Design />
// 		</Provider>
// 	)
// 		.dive({ context: { store } })
// 		.dive();
// 	const instance = wrapper.instance();
// 	const spy = jest.spyOn(instance, 'handleAssignVariable');

// 	instance.forceUpdate();

// 	const Button = wrapper.find('#handleAssignVariable');
// 	Button.simulate('click');
// 	expect(spy).toHaveBeenCalled();
// });

it('deactivateProcess HaveBeenCalled ?', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'deactivateProcess');

	instance.forceUpdate();

	const Button = wrapper.find('#deactivateProcess');
	Button.simulate('click');
	expect(spy).toHaveBeenCalled();
});

it('deleteProcess HaveBeenCalled ?', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'deleteProcess');

	instance.forceUpdate();

	const Button = wrapper.find('#deleteProcess');
	Button.simulate('click');
	expect(spy).toHaveBeenCalled();
});

it('deleteCases HaveBeenCalled ?', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'deleteCases');

	instance.forceUpdate();

	const Button = wrapper.find('#deleteCases');
	Button.simulate('click');
	expect(spy).toHaveBeenCalled();
});

test('exportProcess Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.exportProcess).toBeDefined();
});

test('downloadFile Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.downloadFile).toBeDefined();
});

test('importProject Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.importProject).toBeDefined();
});

test('selectFieldsOfProcess Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.selectFieldsOfProcess).toBeDefined();
});

test('renderSwitch Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.renderSwitch).toBeDefined();
});

test('onSubmit Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.onSubmit).toBeDefined();
});

test('deactivateProcess Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.deactivateProcess).toBeDefined();
});

test('deleteCases Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.deleteCases).toBeDefined();
});

test('handleAssignVariable Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.handleAssignVariable).toBeDefined();
});

test('getAvailableVariables Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getAvailableVariables).toBeDefined();
});

test('getTasks Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getTasks).toBeDefined();
});

test('getAssignedVariables Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.getAssignedVariables).toBeDefined();
});

test('assignVariables Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignVariables).toBeDefined();
});

test('assignVariable Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.assignVariable).toBeDefined();
});

test('unassignVariable Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.unassignVariable).toBeDefined();
});

test('setDefaultVariable Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.setDefaultVariable).toBeDefined();
});

test('setGroupsMembers Function  ', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Design />
		</Provider>
	)
		.dive({ context: { store } })
		.dive();
	const instance = wrapper.instance();
	expect(instance.setGroupsMembers).toBeDefined();
});

describe('actions', () => {
	it('should create an action to insertNew', () => {
		const data = mokeData;
		const expectedAction = {
			type: selectActions.INSERT_NEW,
			data: data,
		};

		expect(selectActions.setInsertNewData(data)).toEqual(expectedAction);
	});

	it('should create an action to getProcessList', () => {
		const data = mokeData;
		const expectedAction = {
			type: selectActions.GET_PROCESS_LIST,
			data: data,
		};

		expect(selectActions.setProcessList(data)).toEqual(expectedAction);
	});

	it('should create an action to getCategories', () => {
		const data = mokeData;
		const expectedAction = {
			type: selectActions.GET_CATEGORIES,
			categories: data,
		};

		expect(selectActions.setCategories(data)).toEqual(expectedAction);
	});

	it('should create an action to deactivateProcess', () => {
		const data = mokeData;
		const expectedAction = {
			type: selectActions.DEACTIVE_PROCESS,
			data: data,
		};

		expect(selectActions.setDeactivateProcess(data)).toEqual(expectedAction);
	});

	it('should create an action to deleteProcess', () => {
		const data = mokeData;
		const expectedAction = {
			type: selectActions.DELETE_PROCESS,
			data: data,
		};

		expect(selectActions.setDeleteProcess(data)).toEqual(expectedAction);
	});
});
