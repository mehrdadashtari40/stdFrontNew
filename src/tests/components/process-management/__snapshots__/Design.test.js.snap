// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , Design Component 1`] = `
<div
  id="content"
>
  <div
    className="row"
  >
    <BigBreadcrumbs
      className="col-xs-12 col-sm-7 col-md-7 col-lg-4"
      icon="fa fa-fw fa-circle-o-notch"
      items={
        Array [
          "فرایند",
          "مدیریت فرآیند",
        ]
      }
    />
  </div>
  <WidgetGrid>
    <div
      className="row"
    >
      <article
        className="col-sm-12 col-md-12 col-lg-12"
      >
        <JarvisWidget
          collapsed={false}
          color={false}
          colorbutton={false}
          custombutton={false}
          deletebutton={false}
          editbutton={false}
          fullscreenbutton={true}
          hidden={false}
          load={false}
          refresh={false}
          sortable={true}
          togglebutton={false}
        >
          <header>
            <span
              className="widget-icon"
            >
              <i
                className="glyphicon glyphicon-edit txt-color-darken"
              />
            </span>
            <h2>
              طراحی فرایند
            </h2>
            <div
              className="progress_btn_bx"
            >
              <button
                className="btn btn_inbox"
                id="handleAssignVariable"
                onClick={[Function]}
              >
                <i
                  className="fa fa-check-square-o"
                />
                <span />
              </button>
              <button
                className="btn btn_inbox"
                id="exportProcess"
                onClick={[Function]}
              >
                <i
                  className="fa fa-download"
                />
                <span />
              </button>
              <a
                className="hidden"
                id="export"
              >
                export file
              </a>
              <button
                className="btn btn_inbox"
                data-target="#importModal"
                data-toggle="modal"
              >
                <i
                  className="fa fa-upload"
                />
                <span />
              </button>
              <button
                className="btn btn_inbox"
                data-target="#myModal"
                data-toggle="modal"
              >
                <i
                  className="fa fa-plus"
                />
                <span />
              </button>
              <button
                className="btn btn_inbox"
                href="#/outlook/folder/paused"
                id="deactivateProcess"
                onClick={[Function]}
              >
                <i
                  className="fa fa-stop"
                />
              </button>
              <button
                className="btn btn_inbox"
                href="#/outlook/folder/paused"
                id="deleteProcess"
                onClick={[Function]}
              >
                <i
                  className="fa fa-close"
                />
              </button>
              <button
                className="btn btn_inbox"
                href="#/outlook/folder/paused"
                id="deleteCases"
                onClick={[Function]}
              >
                <i
                  className="fa fa-ban"
                />
              </button>
            </div>
          </header>
          <div
            className="no-padding"
          >
            <Datatable
              cellSpacing="0"
              className="display projects-table table table-striped table-bordered table-hover"
              filter={true}
              fullTextSearch={false}
              options={
                Object {
                  "buttons": false,
                  "columns": Array [
                    Object {
                      "class": "",
                      "data": null,
                      "defaultContent": "",
                      "orderable": false,
                      "render": [Function],
                    },
                    Object {
                      "class": "row-controller process-designer",
                      "data": "PRO_TITLE",
                    },
                    Object {
                      "class": "row-controller process-designer",
                      "data": "PRO_CATEGORY_LABEL",
                      "render": [Function],
                    },
                    Object {
                      "class": "row-controller process-designer",
                      "data": "PRO_STATUS",
                      "render": [Function],
                    },
                    Object {
                      "class": "row-controller process-designer",
                      "data": "CASES_COUNT",
                    },
                    Object {
                      "class": "row-controller process-designer",
                      "data": "CASES_COUNT_COMPLETED",
                    },
                    Object {
                      "class": "hide id",
                      "data": "PRO_UID",
                    },
                  ],
                  "data": Array [],
                  "iDisplayLength": 10,
                  "order": Array [
                    Array [
                      1,
                      "asc",
                    ],
                  ],
                  "rowID": ".hide.id",
                }
              }
              paginationLength={true}
              width="100%"
            >
              <thead>
                <tr>
                  <th
                    style={
                      Object {
                        "width": "3%",
                      }
                    }
                  />
                  <th
                    className="hasinput"
                  >
                    <input
                      className="form-control"
                      placeholder=""
                      type="text"
                    />
                  </th>
                  <th
                    className="hasinput"
                    style={
                      Object {
                        "width": "15%",
                      }
                    }
                  >
                    <input
                      className="form-control datepicker"
                      data-dateformat="yy/mm/dd"
                      id="dateselect_filter"
                      placeholder=""
                      type="text"
                    />
                  </th>
                  <th
                    className="hasinput"
                    style={
                      Object {
                        "width": "8%",
                      }
                    }
                  >
                    <input
                      className="form-control datepicker"
                      data-dateformat="yy/mm/dd"
                      id="dateselect_filter"
                      placeholder=""
                      type="text"
                    />
                  </th>
                  <th
                    className="hasinput"
                    style={
                      Object {
                        "width": "13%",
                      }
                    }
                  >
                    <input
                      className="form-control"
                      placeholder=""
                      type="text"
                    />
                  </th>
                  <th
                    className="hasinput"
                    style={
                      Object {
                        "width": "13%",
                      }
                    }
                  >
                    <input
                      className="form-control"
                      placeholder=""
                      type="text"
                    />
                  </th>
                </tr>
                <tr>
                  <th />
                  <th />
                  <th />
                  <th />
                  <th />
                  <th />
                </tr>
              </thead>
            </Datatable>
          </div>
        </JarvisWidget>
      </article>
    </div>
  </WidgetGrid>
  <div
    aria-labelledby="myModalLabel"
    className="modal fade"
    id="myModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
            id="myModalLabel"
          />
        </div>
        <div
          className="modal-body"
        >
          <div
            className="smart-form"
          >
            <fieldset>
              <section>
                <label
                  className="input"
                >
                  <i
                    className="icon-append fa fa-question-circle"
                  />
                  <input
                    onChange={[Function]}
                    placeholder=""
                    type="text"
                    value=""
                  />
                  <b
                    className="tooltip tooltip-top-right"
                  >
                    <i
                      className="fa fa-warning txt-color-teal"
                    />
                  </b>
                </label>
              </section>
              <section>
                <label
                  className="textarea"
                >
                  <i
                    className="icon-append fa fa-question-circle"
                  />
                  <textarea
                    onChange={[Function]}
                    placeholder=""
                    rows="3"
                    value=""
                  />
                  <b
                    className="tooltip tooltip-top-right"
                  >
                    <i
                      className="fa fa-warning txt-color-teal"
                    />
                  </b>
                </label>
              </section>
              <section>
                <label
                  className="label"
                  style={
                    Object {
                      "textAlign": "right",
                      "width": "100%",
                    }
                  }
                />
                <label
                  className="select"
                >
                  <select
                    onChange={[Function]}
                    value={0}
                  >
                    <option
                      value="0"
                    />
                  </select>
                  <i />
                </label>
              </section>
            </fieldset>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-default"
            data-dismiss="modal"
            type="button"
          />
          <button
            className="btn btn-primary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="myImportModalLabel"
    className="modal fade"
    id="importModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
            id="myModalLabel"
          />
        </div>
        <div
          className="modal-body"
        >
          <div
            className="smart-form"
          >
            <fieldset>
              <section>
                <input
                  id="project_file"
                  name="project_file"
                  type="hidden"
                />
                <t
                  config={
                    Object {
                      "iconFiletypes": Array [
                        ".pmx",
                      ],
                      "postUrl": "http://ls.arian.co.ir:8081/api/1.0/zums/anrupload/upload?workspace=zums",
                      "showFiletypeIcon": true,
                    }
                  }
                  djsConfig={Object {}}
                  eventHandlers={
                    Object {
                      "success": [Function],
                    }
                  }
                />
              </section>
              <section>
                <label
                  className="label"
                  style={
                    Object {
                      "textAlign": "right",
                      "width": "100%",
                    }
                  }
                />
                <label
                  className="select"
                >
                  <select
                    id="import_option"
                    name="import_option"
                  >
                    <option
                      value="CREATE"
                    />
                    <option
                      value="OVERWRITE"
                    />
                    <option
                      value="DISABLE"
                    />
                    <option
                      value="KEEP"
                    />
                  </select>
                  <i />
                </label>
              </section>
            </fieldset>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-default"
            data-dismiss="modal"
            type="button"
          />
          <button
            className="btn btn-primary"
            id="importBtn"
            onClick={[Function]}
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade-in-up"
    id="assignVariableModal"
    role="dialog"
  >
    <div
      className="modal-dialog modal-full"
    >
      <div
        className="modal-content"
        id="selectFieldBody"
      >
        <div
          className="modal-header"
        >
          <button
            aria-hidden="true"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
          />
        </div>
        <div
          className="modal-body"
        >
          <div
            className="alert alert-danger"
          >
            <p>
              متغیری برای انتصاب وجود ندارد
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade-in-up"
    id="selectProcessTaskModal"
    role="dialog"
  >
    <div
      className="modal-dialog modal-full"
    >
      <div
        className="modal-content"
        id="selectFieldBody"
      >
        <div
          className="modal-header"
        >
          <button
            aria-hidden="true"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            ×
          </button>
          <h4
            className="modal-title"
          >
             زمانبند 
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="alert alert-danger"
          >
            <p>
              متغیری برای انتصاب وجود ندارد
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
